packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -54,6 +60,8 @@ export async function synchronizeCollectedData(
 
     await abortSynchronization({ ...jobContext, reason: err.message });
     throw err;
+  } finally {
+    await eventPublishingQueue.onIdle();
```

Comment:
With how the `eventPublishingQueue` works, `onIdle` should never reject (every operation it performs is wrapped in a try catch), but I suppose it wouldn't hurt to be cautious in case the behavior of the publishing queue changes in the future.